internal/handler/apikey_handler.go

```diff
@@ -111,6 +122,12 @@ func ValidateAPIKey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	authKey, found := a.Cache.Get(apiKeyValue)
+
+	if found {
+		httputil.JSON(w, r, authKey)
+	}
```

Comment:
```suggestion
		httputil.JSON(w, r, authKey)
		return
	}
```

There is a missing return here